src/timesync/timesyncd-manager.c

```diff
@@ -1130,3 +1137,50 @@ int manager_new(Manager **ret) {
 
         return 0;
 }
+
+static int manager_save_time_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        Manager *m = userdata;
+
+        assert(m);
+
+        return manager_save_time_and_rearm(m);
+}
+
+int manager_save_time_add(Manager *m) {
+        int r;
+
```

Comment:
maybe add some asserts here, to document the expected state when this is called, ie. at the very least:

```c
assert(!m->event_save_time);
```